File: programmers/etc/level1/카드_뭉치.py
def solution(cards1, cards2, goal):
    answer = "Yes"
    cards1 = cards1[::-1]
    cards2 = cards2[::-1]
    
    for g in goal:
        if cards1[-1:] == [g]:
            cards1.pop()
        elif cards2[-1:] == [g]:
            cards2.pop()
        else:
            answer = "No"
            break

    return answer


cards1, cards2 = ["i", "drink", "water"], ["want", "to"]
goals = ["i", "want", "to", "drink", "water"]
print(solution(cards1, cards2, goals)) # "Yes"

cards1, cards2 = ["i", "water", "drink"], ["want", "to"]
goals = ["i", "want", "to", "drink", "water"]
print(solution(cards1, cards2, goals)) # "No"

File: Java/programmers/etc/level1/카드뭉치.java
package CodeTest.Java.programmers.etc.level1;

public class 카드뭉치 {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "Yes";

        int n = cards1.length, m = cards2.length;
        int i=0, j=0;
        for (String g : goal) {
            if (i<n && cards1[i].equals(g)) i++;
            else if (j<m && cards2[j].equals(g)) j++;
            else {
                answer = "No";
                break;
            }
        }

        return answer;
    }
}
